### DISCUSSION TITLE
Program plugin - Limit search depth + suffixes for individual folders

### DISCUSSION THREAD

**Discussion by highend** [Flow-Launcher/Flow.Launcher#511] [Ideas]
**Is your feature request related to a problem? Please describe.**

I'm getting too many results (from which I've normally only need one / a few specific ones)
E.g. from a main folder (that may contain a normal .exe which I would like to get listed) but which has subfolders that contain .bat files (which I don't want to be listed).

**Describe the solution you'd like**

I would like two additional features for this plugin.

01. The ability to limit the search depth for an added folder (this must be a "per folder" setting)
E.g.:
```
-1 = Recurse all subfolders
0 = No recursion at all, only files in the root are indexed
1, 2, ... = Additional hierarchy depth to index 
```

02. A black- / whitelist of file suffixes for an added folder (this must be a "per folder" setting)
On the blacklist you would enter those suffixes that should not be indexed (e.g. to exclude ones that are on the global file suffixes list)
On the whitelist you would those that should be indexed (and are not a part of the global file suffixes list)

This would help to e.g. index .bat files in an added folder even if they are not on the global file suffixes list and helps to not index them globally...

**Describe alternatives you've considered**

I don't want to change the organization of the folder structure because for example (.bat) files belong to the application, e.g. because they are used to compile a launcher for it that makes it portable but only the main .exe needs to be indexed.

**Comment by Zeroto521** [comment]
Well, I think Flow is a tool for **quicky** launching something.
If there has something you want to find try to use the Flow backend search engine - **Everything**. 
It may be better in this condition. And **Everything** does well in this way.

**Comment by taooceros** [comment]
Why not simply remove the .bat file from indexing if you don't want them? It is quite easy to do that. @highend

**Comment by jjw24** [comment]
>E.g. from a main folder (that may contain a normal .exe which I would like to get listed) but which has subfolders that contain .bat files (which I don't want to be listed).

@highend I think the bat files are from Explorer plugin? can you shift + enter on the bat file and see if it is Explorer plugin. By default i dont think Program plugin looks for bat files

**Comment by highend** [comment]
The .bat files are my own. They need to stay where they are. I just would like to be able to reduce clutter by defining a max. search depth and an in- / exclusion filter of file types...

**Comment by taooceros** [comment]
Well, I mean you can choose the suffix that will be included to Program plugin. Simply remove the .bat from the setting and reindex will allow you to stop seeing them from querying.




------------------&nbsp;原始邮件&nbsp;------------------


 
The .bat files are my own. They need to stay where they are. I just would like to be able to reduce clutter by defining a max. search depth and an in- / exclusion filter of file types...

**Comment by taooceros** [comment]
![image](https://user-images.githubusercontent.com/45326534/104086707-13c1ba00-5295-11eb-904a-8c23279a4762.png)
Here

**Comment by jjw24** [comment]
I would see this as two parts. The first part is the problem @highend is having, the example given with bat files should be resolved by removing the .bat suffix. Program plugin is for getting programs only, shouldn't be bringing up non-programs, if this is indeed the case we will need more info to troubleshoot.

Second part is the proposed solution for folder search depth and black/whitelisting. This should be something for Explorer plugin, which is dedicated for getting files and folders.
